### DISCUSSION TITLE
can we support automigrate, no need use cli to do that?

### DISCUSSION THREAD

**Discussion by heliping** [SeaQL/sea-orm#1218] [Ideas]
I am a new comer.  the sea-orm is amazing. 

I am familiar with gorm.
in gorm, the migrate is auto, very convenient.
firstly, you can define a struct and table name.
```
type ColumnOriC4 struct {	
	Name       string `gorm:"column:name"`
	Desc       string `gorm:"column:desc"`
}

func (ColumnOriC4) TableName() string {
	return "tab_conf_c4_columnoric4_info"
}
```
secondly, only code like this with db connect.
```
db.AutoMigrate(&ColumnOriC4{})
```

no need cli. when the table changed, the migrate is auto. 
coder no need to care about that.
I think maybe it is a idea to migrate. Thanks.

**Comment by billy1624** [comment]
Hey @heliping, welcome and thanks for the support!! I think this is part of the application logic where user can opt into.

Check this example where all pending migration will be applied before the REST API comes live.

https://github.com/SeaQL/sea-orm/blob/565cb5e6d0794e5af2ed3910e78a3dedf1d683dd/examples/rocket_example/api/src/lib.rs#L140C10-L161

**Reply by tyt2y3** [reply]
Worthy to put on Cookbook

**Reply by liamwh** [reply]
+1 for this request from @heliping. I also come from Go / GORM and decided upon Sea ORM as the best ORM solution, but I miss the [AutoMigrate](https://gorm.io/docs/migration.html) functionality from GORM. The fact that it is missing from Sea ORM means that much more developer time has to be spent writing migration logic, instead of application logic. I would love to see auto-migrate in Sea ORM!

**Reply by liamwh** [reply]
@billy1624, I feel you're misunderstanding the request. With GORM's auto-migrate functionality, you don't need to write your migration code at all. No sqlx equivalent. You just tell GORM to migrate your model, which has GORM struct tags on it, to the SQL database, and GORM does it for you. So in our Rust case, the developer would define the sea_orm attribute-like macros on their own structs, and just tell Sea ORM to automigrate them to the database by referencing the model. See how it is done in GORM [here](https://gorm.io/docs/models.html). In Rust, this could look something like:
```rust
pub struct Post {
    #[sea_orm(primary_key)]
    pub id: i32,
    #[sea_orm(not_null = true)]
    pub title: String,
    #[sea_orm(column_type = "Text")]
    pub text: String,
}

fn main(){
    // Obviously instantiate the database in the correct way, this is just showing a shorthand example of instatiating the DB connection
    let mut db = Db::init()
    db.AutoMigrate(Post)
}
```

**Reply by billy1624** [reply]
Ah, I see what you mean now. You can actually create the schema on the database with [Schema::create_table_from_entity](https://docs.rs/sea-orm/*/sea_orm/schema/struct.Schema.html#method.create_table_from_entity) as docuemented at https://www.sea-ql.org/SeaORM/docs/generate-sea-query-statement/create-table/. However, it's a one-time migration.

Auto migration is no trivial task. At this point we're focusing on stabilizing the API of SeaORM. So, auto migration isn't our utmost priority. But contributions would be very welcome!

**Comment by Huliiiiii** [comment]
Already been implemented in 2.0.
